### PR TITLE
added queue properties parsing in SEND and SUBSCRIBE

### PR DESCRIPTION
**Pull request by priviterag**
added x-message-ttl, x-expires, x-max-length, x-max-length-bytes, x-dead-letter-exchange, x-dead-letter-routing-key and x-max-priority parsing in SEND and SUBSCRIBE

### CODE REVIEW

File: src/rabbit_stomp_util.erl [hunk 1]
```diff
@@ -260,6 +260,26 @@ msg_header_name("1.2") -> ?HEADER_ACK;
...
+                                  true  -> [build_argument(K, V) | Acc];
+                                  false -> Acc
+                                end
+                            end,
+                            [],
+                            Headers),
+    {arguments, Arguments}.
+
+%% build the actual value thru pattern matching
+build_argument(?HEADER_X_DEAD_LETTER_EXCHANGE, Val) -> {list_to_binary(?HEADER_X_DEAD_LETTER_EXCHANGE), longstr, list_to_binary(string:strip(Val))};
```
Comment (michaelklishin) [inline src/rabbit_stomp_util.erl]: Please move head bodies to the next line:

``` erlang
build_argument(?HEADER_X_DEAD_LETTER_EXCHANGE, Val) ->
  {list_to_binary(?HEADER_X_DEAD_LETTER_EXCHANGE), longstr, list_to_binary(string:strip(Val))};
```

File: src/rabbit_stomp_util.erl [hunk 2]
```diff
@@ -260,6 +260,26 @@ msg_header_name("1.2") -> ?HEADER_ACK;
...
+build_argument(?HEADER_X_DEAD_LETTER_ROUTING_KEY, Val) -> {list_to_binary(?HEADER_X_DEAD_LETTER_ROUTING_KEY), longstr, list_to_binary(string:strip(Val))};
+build_argument(?HEADER_X_EXPIRES, Val) -> {list_to_binary(?HEADER_X_EXPIRES), long, list_to_integer(string:strip(Val))};
```
Comment (michaelklishin) [inline src/rabbit_stomp_util.erl]: Expiration is a string in the protocol, [the docs](http://www.rabbitmq.com/ttl.html) mention this.

**Comment by michaelklishin** [comment]
This is submitted for `stable` but clearly was branched off `master` and uses `time_compat` which is present only in `master`.

**Comment by michaelklishin** [comment]
There are no tests for this functionality. It should be fairly easy to test using a STOMP plugin and then `queue.declare` with the expected attributes.

**Comment by michaelklishin** [comment]
I'm closing as this absolutely cannot be merged into stable.

**Comment by priviterag** [comment]
@michaelklishin should I branch from master or stable?

**Comment by michaelklishin** [comment]
Are there any breaking changes? If not, this can go into `3.5.x`.

**Comment by priviterag** [comment]
@michaelklishin for tests, how can I check the properties of a queue from STOMP?

**Comment by michaelklishin** [comment]
@priviterag simply declare the queue with the properties you expect. If they don't match what's already there, you will get a channel exception.

**Comment by michaelklishin** [comment]
@priviterag tests should use both STOMP and the Erlang AMQP 0-9-1 client.

**Pull request closed without merging**